lib/osf-components/addon/components/activity-log/-components/activity-log-display/component.ts

```diff
@@ -0,0 +1,241 @@
+}
+
+export default class ActivityLogDisplayComponent extends Component<ActivityLogDisplayArgs> {
+    @service intl!: Intl;
+    @tracked activityDisplay = '';
+    @tracked isLoading = true;
+    @tracked log!: LogModel;
+    user!: UserModel;
+    linkedNode!: NodeModel;
+    linkedRegistration!: RegistrationModel;
+    node!: NodeModel;
+
+    /**
+     * constructor
+     *
+     * @param owner The owner
+     * @param args The injected args
+     */
+    constructor(owner: unknown, args: ActivityLogDisplayArgs) {
+        super(owner, args);
+        this.log = args.log;
+        taskFor(this.loadModels).perform();
+    }
+
+    /**
+     * buildAHrefElement
+     *
+     * @description Abstracted method to build an "a" element
+     *
+     * @param url The url for the href
+     * @param value The text to appear as clickable
+     * @returns An ahref formatted string
+     */
+    private buildAHrefElement(url: string | undefined, value: string): string {
+        url = url || '';
+        return `<a href="${this.toRelativeUrl(url)}">${value}</a>`;
+    }
+
+    /**
+     * Utility function to convert absolute URLs to relative urls
+     * See:
+     *      http://stackoverflow.com/questions/736513/how-do-i-parse-a-url-into-hostname-and-path-in-javascript
+     * This method have no effect on external urls.
+     * @param url {string} url to be converted
+     * @returns {string} converted relative url
+     */
+    private toRelativeUrl(url: string): string {
+        const parser = document.createElement('a');
+        parser.href = url;
+        let relative_url = url;
+        if (window.location.hostname === parser.hostname){
+            relative_url = parser.pathname + parser.search + parser.hash;
+        }
+        return relative_url;
+    }
+
+    /**
+     * buildParam
+     *
+     * @description Abstracted method to build (assembly) all of the necessary translation variables
+     *
+     * @returns A Param model
+     */
+    private buildParam(): ParamModel {
+        return {
+            anonymousLink: this.log.params.anonymousLink ? 'an anonymous' : 'a',
+            fullName:  this.buildFullNameUrl(),
+            license: this.log.params.license,
+            node:  this.buildNodeUrl(),
+            path: this.log.params.path,
+            pathType: this.log.params.pathType,
+            tag: this.log.params.tag,
+            pointer: this.getPointer(),
+            pointerCategory: this.getPointerCategory(),
+        };
+    }
+
+    /**
+     * loadModels
+     *
+     * @description Hydrates all the models before displaying the component
+     *
+     * @returns a void promise
+     */
+    @task
+    @waitFor
+    private async loadModels(): Promise<void> {
+        this.user = await this.log.user;
+        this.linkedNode = await this.log.linkedNode;
+        this.linkedRegistration = await this.log.linkedRegistration;
+        this.node = await this.log.node;
+        this.isLoading = false;
```

Comment:
For future reference: instead of manually setting a flag like this, you can just use `this.loadModels.isRunning` if you are waiting for a `task` to finish